lib/actions/api.js

```diff
@@ -570,11 +570,13 @@ export function findStopTimesForStop (params) {
       queryParams.startTime = nowInSeconds
     }
 
-    // Clear existing stop times to prevent extra rendering of stop viewer while schedule is
+    // If reuqested, clear existing stop times to prevent extra rendering of stop viewer while schedule is
     // fetched and a schedule for the same date/stop was shown before. This also has the nice side
     // effect to force a render, initially showing the top row, after changing the schedule view date
     // and all schedules for the new date are the same as the previous date.
-    dispatch(findStopTimesForStopResponse({ stopId, stopTimes: [] }))
+    if (clearData) {
+      dispatch(findStopTimesForStopResponse({ stopId, stopTimes: [] }))
+    }
```

Comment:
I didn't notice this before, but this doesn't quite seem like the correct approach. I'd prefer that we send off an action to indicate that we are in a fetching state. The component itself can then read the fetch state from the reducer and determine what to render (rather than making the render behavior stem from the presence/absence of stop data). Currently, changing the date still has the effect of showing a "no stop times" message rather than a loading spinner.